yorkie/backend/db/doc_info.go

```diff
@@ -48,3 +48,9 @@ func (info *DocInfo) GetKey() (*key.Key, error) {
 
 	return docKey, nil
 }
+
+// ShallowCopy creates a shallow copy of this DocInfo.
+func (info *DocInfo) ShallowCopy() *DocInfo {
+	clone := *info
+	return &clone
+}
```

Comment:
Looking at the properties of `DocInfo` and the logic of this function, it seems that this function is doing a deep copy.
This function is used in `FindDocInfoByKey()` in `yorkie/backend/db/memory/db.go`. Is there any intention here that it should be a shallow copy?
If there is an intention, it seems to be necessary to verify the case where a problem can occur when it is not a shallow copy.

Reply:
I forgot to add a comment.

In MemDB, when retrieving objects, references to stored objects are returned instead of new objects returned. 
This causes problems when directly modifying the imported object.

Reply:
Thanks for the explanation.
So, I think the function name should be DeepCopy rather than ShallowCopy, what do you think?

Reply:
I don't think a DeepCopy will work with that code, can you please explain why it is?

Reply:
I replace ShallowCopy with DeepCopy because using ShallowCopy seems to have a higher chance of making a mistake. Thanks for the good comments.